### PR TITLE
feat: [DHIS2-15906] Add form features for relationship

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
This PR adds:
- Show first page in registration form
- Attribute option combos
- Autogenerateable events

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/TrackedEntityRelationshipsWrapper/hooks/useDataEntryReduxConverter.js [hunk 1]
```diff
@@ -96,31 +98,68 @@ export const useDataEntryReduxConverter = ({
...
+        const attributeCategoryOptionsId = 'attributeCategoryOptions';
+        const attributeCategoryOptions = Object.keys(serverValuesForMainValues)
+            .filter(key => key.startsWith(attributeCategoryOptionsId))
+            .reduce((acc, key) => {
+                const categoryId = key.split('-')[1];
+                acc[categoryId] = serverValuesForMainValues[key];
+                return acc;
+            }, {});
+
+        const formServerValues = serverValuesForFormValues[Section.groups.ENROLLMENT];
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/TrackedEntityRelationshipsWrapper/hooks/useDataEntryReduxConverter.js:128]: Hey,
I am not able to save the TEI attributes. I dug a bit at it looks like `serverValuesForFormValues[Section.groups.ENROLLMENT]` is `undefined`. Can you have a look?

https://github.com/dhis2/capture-app/assets/6817073/c597c80d-e216-4228-a9a5-ca2c9b026afb


Thanks!

File: src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/RegisterTei/RegisterTei.container.js [hunk 2]
```diff
@@ -14,26 +13,36 @@ export const RegisterTei = ({
     suggestedProgramId,
 }: ContainerProps) => {
     const dataEntryId = 'relationship';
-    const itemId = useSelector(({ dataEntries }) => dataEntries[dataEntryId]?.itemId);
+    // const itemId = useSelector(({ dataEntries }) => dataEntries[dataEntryId]?.itemId);
     const error = useSelector(({ newRelationshipRegisterTei }) => (newRelationshipRegisterTei.error));
     const selectedScopeId = suggestedProgramId || trackedEntityTypeId;
     const { trackedEntityName } = useScopeInfo(selectedScopeId);
-    const { buildTeiPayload } = useDataEntryReduxConverter({
+    /* const {
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/common/TEIRelationshipsWidget/RegisterTei/RegisterTei.container.js:20]: There are some commented lines of code in this file. Can you clean them before merging? Thanks!

**Review by simonadomnisoru — APPROVED** [review]
LGTM! I just added a reminder to clean up commented lines of code.

File: src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/hooks/useBuildEnrollmentPayload.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    const formValues = useSelector(({ formsValues }) => formsValues[dataEntryKey]);
+    const dataEntryFieldValues = useSelector(({ dataEntriesFieldsValue }) => dataEntriesFieldsValue[dataEntryKey]);
+    const dataEntryFieldsMeta = useSelector(({ dataEntriesFieldsMeta }) => dataEntriesFieldsMeta[dataEntryKey]);
+    const { formFoundation: scopeFormFoundation } = useMetadataForRegistrationForm({ selectedScopeId: programId });
+    const { firstStageMetaData } = useBuildFirstStageRegistration(programId);
+    const { formFoundation } = useMergeFormFoundationsIfApplicable(scopeFormFoundation, firstStageMetaData);
+
+    const buildTeiWithEnrollment = (): EnrollmentPayload => {
+        // $FlowFixMe - Business logic dictates that formFoundation is not null in this callback
+        if (!formFoundation) return null;
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/hooks/useBuildEnrollmentPayload.js:111]: Thought about this a bit again, and I think it's better to just throw in situations where logic dictates some data to be there. This way we will not have any unintended consequences. 

```suggestion
	if (!formFoundation) throw Error('form foundation object not found');
```

File: src/core_modules/capture-core/components/DataEntries/TeiRegistrationEntry/hooks/useBuildTeiPayload.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    itemId = 'newTei',
+    orgUnitId,
+}: Props) => {
+    const dataEntryKey = getDataEntryKey(dataEntryId, itemId);
+    const { formFoundation } = useMetadataForRegistrationForm({ selectedScopeId: trackedEntityTypeId });
+    const formValues = useSelector(({ formsValues }) => formsValues[dataEntryKey]);
+
+    const buildTeiWithoutEnrollment = (): TeiPayload => {
+        // $FlowFixMe - Business logic dictates that formFoundation is defined
+        if (!formFoundation) return null;
```
Comment (JoakimSM) [inline src/core_modules/capture-core/components/DataEntries/TeiRegistrationEntry/hooks/useBuildTeiPayload.js:53]:
```suggestion
	if (!formFoundation) throw Error('form foundation object not found');
```

**Pull request merged**